Pyrado/pyrado/utils/experiments.py

```diff
@@ -67,24 +74,176 @@ def load_experiment(
 
     :param ex_dir: experiment's parent directory
     :param args: arguments from the argument parser, pass `None` to fall back to the values from the default argparser
-    :return: environment, policy, and (optional) algorithm-specific output, e.g. value function
+    :return: environment, policy, and optional output (e.g. valuefcn)
     """
     env, policy, extra = None, None, dict()
 
     if args is None:
-        # Fall back to default arguments. By passing [], we ignore the command line arguments.
+        # Fall back to default arguments. By passing [], we ignore the command line arguments
         args = get_argparser().parse_args([])
 
     # Hyper-parameters
     extra["hparams"] = load_hyperparameters(ex_dir)
 
-    # Algorithm, environment, policy, and more
-    algo = pyrado.load("algo.pkl", ex_dir)
-    env, policy, extra = algo.load_snapshot(args)
+    # Algorithm specific
+    algo = Algorithm.load_snapshot(load_dir=ex_dir, load_name="algo")
+
+    if algo.name == "spota":
```

Comment:
This is weird, why did you add it again?

Reply:
Merging error…